page_iterator.go

```diff
@@ -87,6 +89,10 @@ func (pI *PageIterator[T]) Iterate(context context.Context, callback func(pageIt
 			return nil
 		}
 
+		if pI.currentPage.getOdataNextLink() == nil || *pI.currentPage.getOdataNextLink() == "" {
```

Comment:
This change allows the iterator to maintain `currentPage` so that the new getters can return the correct values. Previously it was being overwritten.